packages/hash/frontend/src/blocks/page/Comments/CommentButton.tsx

```diff
@@ -0,0 +1,64 @@
+import { FunctionComponent, useCallback, useRef } from "react";
+import { IconButton, FontAwesomeIcon } from "@hashintel/hash-design-system";
+import { faComment } from "@fortawesome/free-regular-svg-icons";
+import Box from "@mui/material/Box";
+import { TextToken } from "@hashintel/hash-shared/graphql/types";
+import { useBlockView } from "../BlockViewContext";
+import { createCommentPluginKey } from "../createCommentPlugin";
+import { useCreateComment } from "../../../components/hooks/useCreateComment";
+import { useRouteAccountInfo } from "../../../shared/routing";
+
+type CommentButtonProps = {
+  blockId: string | null;
+  className: string;
+};
+
+export const CommentButton: FunctionComponent<CommentButtonProps> = ({
+  blockId,
+  className,
+}) => {
+  const { accountId } = useRouteAccountInfo();
+  const blockView = useBlockView();
+  const containerRef = useRef<HTMLButtonElement>(null);
+
+  const [createComment] = useCreateComment(accountId);
+
+  const submitComment = useCallback(
+    async (content: TextToken[]) => {
+      if (blockId) {
+        await createComment(blockId, content);
+      }
+    },
+    [createComment, blockId],
+  );
+
+  const openCommentInput = () => {
+    const view = blockView.editorView;
+    const { tr } = view.state;
+    tr.setMeta(createCommentPluginKey, {
+      type: "open",
+      payload: {
+        anchorNode: containerRef.current,
+        blockId,
+        onSubmit: submitComment,
+      },
+    });
+    view.dispatch(tr);
+  };
+
+  return (
+    <Box ref={containerRef} className={className}>
+      <IconButton
+        onClick={openCommentInput}
+        sx={{
+          padding: 0.5,
+          borderRadius: 1,
+          opacity: blockView.hovered ? 1 : 0,
```

Comment:
This is the same mechanism that's used for hiding/showing the `BlockHandle`. It re-renders because we re-render the BlockView whenever `hovered` is updated (see `createHoverHandler` in `BlockView.tsx`).

Reply:
That's an implementation detail of `BlockView`, though. `BlockHandle` also should not be relying on this. I'm not sure I don't understand why we don't just use CSS for this.

Reply:
Addressed this in 492aace973946f98342f43dcc1ee6898fe871d24.